### PR TITLE
Remove dead code from daemon/daemon.go

### LINKED ISSUE #12492: Remove dead code from daemon/daemon.go
In `*Daemon.register`:
```
			var err error
			cmd.ProcessConfig.Process, err = os.FindProcess(existingPid)
			if err != nil {
				logrus.Debugf("cannot find existing process for %d", existingPid)
			}
```
is dead code. We can rely on `container.Pid` without process start time anyway for killing processes.

### PR DESCRIPTION
**Pull request by jmzwcn**
fix #12492
Signed-off-by: Daniel Zhang jmzwcn@gmail.com

### CODE REVIEW

File: daemon/daemon.go [hunk 1]
```diff
@@ -237,11 +236,6 @@ func (daemon *Daemon) register(container *Container, updateSuffixarray bool) err
 			cmd := &execdriver.Command{
 				ID: container.ID,
 			}
-			var err error
-			cmd.ProcessConfig.Process, err = os.FindProcess(existingPid)
```
Comment (runcom) [inline daemon/daemon.go]: don't we have to assign `cmd.ProcessConfig.Process = containerPid` here?

File: daemon/daemon.go [hunk 2]
```diff
@@ -237,11 +236,6 @@ func (daemon *Daemon) register(container *Container, updateSuffixarray bool) err
...
-			if err != nil {
-				logrus.Debugf("cannot find existing process for %d", existingPid)
-			}
 			daemon.execDriver.Terminate(cmd)
```
Comment (cpuguy83) [inline daemon/daemon.go]: I think we should check error and log as debug here like we were above... @LK4D4 ?

Comment (LK4D4) [inline daemon/daemon.go]: I think it is okay. `Terminate` will return error in 99% cases.

**Comment by LK4D4** [comment]
LGTM

**Comment by cpuguy83** [comment]
ok, LGTM.

**Pull request merged**